frontend/public/components/operator-lifecycle-manager/create-operand.tsx

```diff
@@ -18,49 +21,145 @@ import {
   referenceFor,
   referenceForModel,
   Status,
+  nameForModel,
+  CustomResourceDefinitionKind,
 } from '../../module/k8s';
 import { ClusterServiceVersionKind, referenceForProvidedAPI, providedAPIsFor, CRDDescription, ClusterServiceVersionLogo, APIServiceDefinition } from './index';
-import { ClusterServiceVersionModel } from '../../models';
+import { ClusterServiceVersionModel, CustomResourceDefinitionModel } from '../../models';
 import { Firehose } from '../utils/firehose';
-import { NumberSpinner, StatusBox, BreadCrumbs, history, SelectorInput, ListDropdown, AsyncComponent } from '../utils';
+import { NumberSpinner, StatusBox, BreadCrumbs, history, SelectorInput, ListDropdown, AsyncComponent, resourcePathFromModel, FirehoseResult, useScrollToTopOnMount } from '../utils';
 import { SpecCapability, StatusCapability, Descriptor } from './descriptors/types';
 import { ResourceRequirements } from './descriptors/spec/resource-requirements';
 import { RootState } from '../../redux';
 import { CreateYAML } from '../create-yaml';
 import { RadioGroup } from '../radio';
 import { ConfigureUpdateStrategy } from '../modals/configure-update-strategy-modal';
-import { NodeAffinity, PodAffinity } from './descriptors/spec/affinity';
+import { NodeAffinity, PodAffinity, defaultNodeAffinity, defaultPodAffinity } from './descriptors/spec/affinity';
 
 const annotationKey = 'alm-examples';
 
+enum Validations {
+  maximum = 'maximum',
+  minimum = 'minimum',
+  maxLength = 'maxLength',
+  minLength = 'minLength',
+  pattern = 'pattern',
+}
+
 /**
  * Combines OLM descriptor with JSONSchema.
  */
 type OperandField = {
   path: string;
   displayName: string;
-  description: string;
-  type: string;
+  description?: string;
+  type: JSONSchema6TypeName;
+  required: boolean;
+  validation: {
+    [Validations.maximum]?: number;
+    [Validations.minimum]?: number;
+    [Validations.maxLength]?: number;
+    [Validations.minLength]?: number;
+    [Validations.pattern]?: string;
+  };
   capabilities: (SpecCapability | StatusCapability)[];
 };
 
+type FieldErrors = {
+  [path: string]: string;
+};
+
 const fieldsFor = (providedAPI: CRDDescription) => _.get(providedAPI, 'specDescriptors', [] as Descriptor[]).map(desc => ({
   path: desc.path,
   displayName: desc.displayName,
   description: desc.description,
-  // TODO(alecmerdler): Grab the type from JSONSchema, but could need to deep access `properties` in order to get to it...
   type: null,
+  required: false,
+  validation: null,
   capabilities: desc['x-descriptors'],
 })) as OperandField[];
 
+const fieldsForOpenAPI = (crd: CustomResourceDefinitionKind): OperandField[] => {
+  const openAPIV3Schema = _.get(crd, 'spec.validation.openAPIV3Schema', {});
+
+  if (_.isEmpty(openAPIV3Schema)) {
+    return [];
+  }
+
+  const fields: OperandField[] = _.flatten(_.map(openAPIV3Schema.properties.spec.properties, (val, key: string) => {
```

Comment:
Can any of the fields in this path be null? Looking at the CRD API definition, they're not marked required. I'd throw in a `_.get` here just in case.